genesis/camino.go

```diff
@@ -34,16 +34,24 @@ func validateCaminoConfig(config *Config) error {
 			return errors.New("deposit minimum duration is greater than maximum duration")
 		}
 
-		if offer.MinDuration < offer.UnlockHalfPeriodDuration {
+		if offer.MinDuration < offer.NoRewardsPeriodDuration {
```

Comment:
Suggestion:
Both offer.NoRewardsPeriodDuration and offer.UnlockPeriodDuration are based on MaxDuration

The final calculation will then be:
txNoRewardDuration = (offer.NoRewardsPeriodDuration * tx.Duration) / offer.MaxDuration
With that the next 2 checks can be removed.

Pls. also add a flags bitfield (32/64 bit) into DepositOffer (initially 0), so we can change the logic later easily.

Reply:
but that means that we won't be able to set strict UnlockPeriodDuration in offer, right? It will be dynamic, based on deposit duration specified by user?